frb_example/benches/dart/lib/interceptor.web.dart

```diff
@@ -0,0 +1,76 @@
+/// interceptor (on web platform)
+///
+/// underlying timer is based on [window.performance](https://api.dart.dev/stable/2.18.2/dart-html/Window/performance.html)
+/// a.k.a [Performance](https://developer.mozilla.org/en-US/docs/Web/API/Performance)
+///
+/// mostly a lot of boilerplate so that dart agrees to compile.
```

Comment:
Totally agree a lot of boilerplate. Seems that we can remove them. The simple idea is that, we just extract the common part into a single code file + only write down the really different part between io and web in the .io.dart and .web.dart. Guess that there is only minimal differences. Maybe only the "start" and "stop"?

Reply:
Isn't it what I've done already ? There's the platform-agnostic with all the generic logic and a `.io.dart` for `io` only and `.web.dart` for `web`.

Reply:
in my current implementation logic is abstracted away under generics in `interceptor.dart`, while both `io.dart` and `web.dart` are just there to concretize the types mostly.

for example `windows.performance` doesn't exist on dart vm, AFAIK, same with `Stopwatch` which doesn't exist on dart2js, so on and so forth.

Reply:
Maybe the whole file:

```dart
// common
...
class TimeWatch {
  void start();
  int end(); // return elapse time
}

// .io.dart

class TimeWatchImpl extends TimeWatch {
  final _stopwatch = Stopwatch();

  @override
  void start() {
   _stopwatch.start();
  }

  @override
  void stop() {
   _stopwatch.stop();
   return elapsedMicroseconds;
  }
}

// .web.dart

class TimeWatchImpl implements TimeWatch {
  int? _starts;
  void start() {
    _starts = window.performance.now().round();
  }

  @override
  void stop() {
    return window.performance.now().round() - _starts;
  }
}
```

As for UniqueTimeWatch:

```
class UniqueTimeWatch  {
  final uuid = Uuid.v4();
  final inner = TimeWatchImpl(); // btw this is conditionally imported from io or web
}
```

etc

Reply:
I don't see any difference with what I already did, can you explain ?

Reply:
Ah maybe I get what you mean now. I have to look into it indeed.